source/plugin/basics/injection.rst

```diff
@@ -0,0 +1,178 @@
+
+Temporary List of Injected Types
+================================
+
+* Game
+
+    * The ``Game`` object is the core accessor of the SpongeAPI.
+
+* PluginManager
+
+    * Manages the plugins loaded by the implementation. Can retrieve another plugin's ``PluginContainer``.
+
+* EventManager
+
+    * Manages the registration of event handlers and the dispatching of events.
+
+* GameRegistry
+
+    * Provices an easy way to retrieve types from a ``Game``.
+
+* PluginContainer
+
+    * A ``@Plugin`` class wrapper, used to retrieve information from the annotation for easier use.
+
+* Logger
+
+    * Used to identify the plugin from which logged messages are sent.
+
+* File (Has to have an additional annotation specifying specific file)
+
+    * Currently, specifications are ``ConfigFile`` and ``ConfigDir``.
+
+* ConfigDir
+
+    * Used to inject the plugin's configuration directory: ``./mods/`` OR ``./mods/<Plugin#id>/`` depending on ``sharedRoot``
+
+* ConfigFile
+
+    * Used to inject the plugin's specific configuration file: ``<Plugin#id>.conf``
+
+Injection Examples
+==================
+
+There are a few references which are difficult to get - or, in some cases, impossible - without injection. While these
+may not be absolutely vital to every plugin, they're quite frequently used.
+
+.. note::
+
+    Note that it's *almost always* recommended to inject your objects within the main class, as it's instantiated with the Guice injector when the plugin is loaded.
+
+Logger
+~~~~~~
+
+.. tip::
+
+    View `Logging and Debugging <../basics/logging.rst>`_ for a complete guide, specifically for the Logger.
+
+Game
+~~~~
+
+The ``Game`` object is the opening for many of the internal functions of the SpongeAPI, from the ``EventManager`` to the
+``Server`` and even the Sync/Async ``Scheduler``.
+
+It's entirely possible to recieve the ``Game`` object from within most events, however it is commonly obtained through
+an injection.
+
+**Example - Field**
+
+.. code-block:: java
+
+    import com.google.inject.Inject;
+    import org.spongepowered.api.Game;
+
+    @Inject
+    private Game game;
+
+**Example - Method**
+
+.. code-block:: java
+
+    import com.google.inject.Inject;
+    import org.spongepowered.api.Game;
+
+    private Game game;
+
+    @Inject
+    private void setGame(Game game) {
+        this.game = game;
+    }
+
+**Example - Constructor**
+
+.. code-block:: java
+
+    import com.google.inject.Inject;
+    import org.spongepowered.api.Game;
+
+    // For the purpose of this tutorial, "Apple" is the class name.
+
+    private Game game;
+
+    @Inject
+    public Apple(Game game) {
+        this.game = game;
+    }
+
+Config Directory
+~~~~~~~~~~~~~~~~
+
+This is a big one, especially if you plan on having more than a single file stored for your plugin to later call.
+Thankfully, with the power of the Guice injector, it's never been easier to get your directory.
```

Comment:
We're not trying to advertise here, we've already sold people on "how easy it is" to make Sponge plugins... The goal is to describe features in an unbiased tone